### PR TITLE
[pytest] Add new test to check BGP update time

### PR DESCRIPTION
**Pull request by lolyu**
### Description of PR

Summary:
Fixes # (issue)

Add a testcase `test_bgp_update_timer` to cover the following testing scenario:
1. For FRRrouting, ensure SONiC sends BGP updates to neighbors after receiving within 1s.
2. For Quagga, ensure the time interval is within 20s.

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### How did you do it?
1. Start two new BGP sessions.
2. Send BGP update to announce route from the first neighbor to DUT.
3. Ensure DUT sends a BGP update to the second neighbor to announce the received route.
4. Time the interval between Step 2 and Step 3.
5. Send BGP update to withdraw the route announced in Step 2 from the first neighbor to DUT.
6. Ensure DUT sends a BGP update to the second neighbor to withdraw the route.
7. Time the interval between Step 5 and Step 6.
8. Repeat Step 2 to Step 7 for 5 times.
9. Ensure the medians of announce update intervals and withdraw updates intervals are within threshold. 

#### How did you verify/test it?
1. Test it against `t0` 
```
bgp/test_bgp_update_timer.py::test_bgp_update_timer PASSED                                                                                                                  [100%]
```
2. Test it against `t1`
```
bgp/test_bgp_update_timer.py::test_bgp_update_timer PASSED                                                                                                                  [100%]
```
3. Test it over 201811
```
bgp/test_bgp_update_timer.py::test_bgp_update_timer PASSED                                                                                                                  [100%]
```

#### Supported testbed topology if it's a new test case?
`t0` and `t1`

### CODE REVIEW

**Comment by lolyu** [comment]
Retest this please

File: tests/bgp/test_bgp_update_timer.py [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+
+class BGPNeighbor(object):
+
+    def __init__(self, duthost, ptfhost, name, iface,
+                 neighbor_ip, neighbor_asn,
+                 dut_ip, dut_asn, port, is_quagga=False):
+        self.duthost = duthost
+        self.ptfhost = ptfhost
+        self.ptfip = ptfhost.host.options["inventory_manager"].get_host(
+            ptfhost.hostname).vars["ansible_host"]
```
Comment (wangxin) [inline tests/bgp/test_bgp_update_timer.py:53]: You can just use `ptfhost.mgmt_ip` to get the PTF management IP address.

File: tests/bgp/test_bgp_update_timer.py [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+NEIGHBOR_ASN1 = 61001
+NEIGHBOR_PORT0 = 11000
+NEIGHBOR_PORT1 = 11001
+
+
+def _write_variable_from_j2_to_configdb(duthost, template_file, **kwargs):
+    save_dest_path = kwargs.pop("save_dest_path", "/tmp/temp.j2")
+    keep_dest_file = kwargs.pop("keep_dest_file", False)
+    duthost.host.options["variable_manager"].extra_vars.update(kwargs)
+    duthost.template(src=template_file, dest=save_dest_path)
```
Comment (wangxin) [inline tests/bgp/test_bgp_update_timer.py:39]: Can you use the jinja2.Template and duthost.copy module to generate json config file? This is more explicit than updating extra vars of duthost's variable_manager.

Example:
https://github.com/Azure/sonic-mgmt/blob/master/tests/bgp/test_bgp_bbr.py#L44

**Comment by lolyu** [comment]
Retest this please

File: tests/bgp/test_bgp_update_timer.py [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+
+        if self.is_quagga:
+            allow_ebgp_multihop_cmd = (
+                "vtysh "
+                "-c 'configure terminal' "
+                "-c 'router bgp %s' "
+                "-c 'neighbor %s ebgp-multihop'"
+            )
+            allow_ebgp_multihop_cmd %= (self.peer_asn, self.ip)
+            self.duthost.shell(allow_ebgp_multihop_cmd)
```
Comment (wangxin) [inline tests/bgp/test_bgp_update_timer.py:110]: Need to undo this change on DUT after test?

Comment (lolyu) [reply tests/bgp/test_bgp_update_timer.py:110]: this is for the new neighbors, after test, those new neighbors are removed.

File: tests/bgp/test_bgp_update_timer.py (hunk 1 above)
Comment (lolyu) [reply tests/bgp/test_bgp_update_timer.py:53]: Nice to know, will update, thanks!

File: tests/bgp/test_bgp_update_timer.py (hunk 2 above)
Comment (lolyu) [reply tests/bgp/test_bgp_update_timer.py:39]: Sure, will update.

**Comment by lolyu** [comment]
Rest passed, @wangxin, please help to review again, thanks.

File: ansible/library/exabgp.py [hunk 4]
```diff
@@ -55,7 +55,7 @@
     try:
         line = stdin.readline()
         obj = json.loads(line)
-        f = open("/tmp/exabgp-" + obj["neighbor"]["ip"], "a")
+        f = open("/tmp/exabgp-" + obj["neighbor"]["address"]["local"], "a")
```
Comment (wangxin) [inline ansible/library/exabgp.py:58]: Why make this change?

Comment (lolyu) [reply ansible/library/exabgp.py:58]: Because DUT uses loopback address as router ID for both sessions, those two `exabgp` will dump to the same file.

**Review by wangxin — APPROVED** [review]

**Review by bingwang-ms — APPROVED** [review]
